lib/IRGen/GenEnum.cpp

```diff
@@ -2322,6 +2328,19 @@ namespace {
     }
 
   public:
+    StackAddress allocateStack(IRGenFunction &IGF, SILType T,
+                               const llvm::Twine &name) const override {
+      assert(TIK < Fixed);
+
+      if (ElementsWithNoPayload.size() > 1)
+        return EnumImplStrategy::allocateStack(IGF, T, name);
```

Comment:
You could handle this for all cases by just adding 4, I think.

Reply:
IIRC, if we can't use extra inhabitants then we only use a separate discriminator.  So if you want to actually do the size computation correctly, it's not very much harder to just do it right for all discriminator sizes.

But yeah, I think it's probably better to avoid the computation anyway.